ci/test-conformance-eks.sh

```diff
@@ -0,0 +1,228 @@
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+function setup_eks() {
+
+    echo "=== This cluster to be created is named: ${CLUSTER} ==="
+    echo "CLUSTERNAME=${CLUSTER}" > ci_properties.txt
+
+    echo "=== Using the following awscli version ==="
+    aws --version
+
+    set +e
+    aws configure << EOF
+${AWS_ACCESS_KEY}
+${AWS_SECRET_KEY}
+${REGION}
+JSON
+EOF
+    set -e
+    printf "\n"
+    echo "=== Using the following kubectl ==="
+    which kubectl
+
+    echo '=== Creating a cluster in EKS ==='
+    eksctl create cluster \
+      --name ${CLUSTER} --region ${REGION} --version=${K8S_VERSION} \
+      --nodegroup-name workers --node-type ${AWS_NODE_TYPE} --nodes 2 \
+      --ssh-access --ssh-public-key ${SSH_KEY_PATH} \
+      --managed
+    if [[ $? -ne 0 ]]; then
+        echo "=== Failed to deploy EKS cluster! ==="
+        exit 1
+    fi
+
+    mkdir -p ${WORKSPACE}/jenkins/out/
+    eksctl utils write-kubeconfig \
+      --cluster=${CLUSTER} --kubeconfig=${KUBECONFIG_PATH}/kubeconfig
+    export KUBECONFIG="${KUBECONFIG_PATH}/kubeconfig"
+
+    sleep 5
+    if [[ $(kubectl get svc) ]]; then
+        echo "=== EKS cluster setup succeeded ==="
+    else
+        echo "=== EKS kubectl is not configured correctly! ==="
+        exit 1
+    fi
+}
+
+function deliver_antrea_to_eks() {
+
+    echo "=== Configuring Antrea for cluster ==="
+
+    # EKS service CIDR is currently assigned by AWS,
+    # either from 172.20.0.0/16 or 10.100.0.0/16 depending on the range of VPC
+    # https://forums.aws.amazon.com/thread.jspa?messageID=859958
+    k8s_svc_addr=$(kubectl get svc -A | grep kubernetes | awk '{print $4}')
+
+    if [[ $k8s_svc_addr == 10.100.* ]]; then
+    	 k8s_svc_cidr='10.100.0.0/16'
+    elif [[ $k8s_svc_addr == 172.20.* ]]; then
+    	 k8s_svc_cidr='172.20.0.0/16'
+    else
+    	 echo "Cannot determine EKS serviceCIDR!"
+         exit 1
+    fi
+
+    set +e
+    worker_node_ip_1=$(kubectl get nodes -o wide | sed -n '2 p' | awk '{print $7}')
+    node_mtu=$(ssh -o StrictHostKeyChecking=no -l "ec2-user" $worker_node_ip_1 \
+      'export PATH=$PATH:/usr/sbin; ip a | grep -E eth0.*mtu | cut -d " " -f5')
+
+    sed -i "s|#defaultMTU: 1450|defaultMTU: ${node_mtu}|g" ../build/yamls/antrea-eks.yml
+    sed -i "s|#serviceCIDR: 10.96.0.0/12|serviceCIDR: ${k8s_svc_cidr}|g" ../build/yamls/antrea-eks.yml
+    echo "defaultMTU set as ${node_mtu}"
+    echo "seviceCIDR set as ${k8s_svc_cidr}"
+
+    kubectl apply -f ../build/yamls/antrea-eks.yml
+    set -e
+
+    kubectl rollout status --timeout=2m deployment.apps/antrea-controller -n kube-system
+    kubectl rollout status --timeout=2m daemonset/antrea-agent -n kube-system
+
+    echo "=== Antrea has been deployed for EKS cluster ${CLUSTER} ==="
+}
+
+function run_conformance() {
+    echo "=== Running Antrea Conformance and Network Policy Tests ==="
+
+    chmod +x run-k8s-e2e-tests.sh
```

Comment:
the file is already executable as far as I can tell.
is it required for jenkins? if yes could we do it in the yaml shell?

Reply:
Let me remove that. I still remember a permission issue when I run this script in Jenkins. But for the purpose of this script I agree that we don't need this